code/game/objects/machinery/mortar.dm

```diff
@@ -223,11 +223,11 @@
 	to_chat(user, "<span class='notice'>You disconnect the [binocs] from their linked mortar.")
 
 ///Start firing the gun on target and increase tally
-/obj/machinery/deployable/mortar/proc/begin_fire(target, obj/item/mortal_shell/arty_shell)
+/obj/machinery/deployable/mortar/proc/begin_fire(atom/target, obj/item/mortal_shell/arty_shell)
 	firing = TRUE
-	for(var/mob/M in GLOB.player_list)
-		if(get_dist(M , src) <= 7)
-			shake_camera(M, 1, 1)
+	for(var/mob/M in (cheap_get_humans_near(src, 7) + cheap_get_xenos_near(src,7)))
+		shake_camera(M, 1, 1)
+
```

Comment:
I dont think this is cheaper as cheap_get_humans_near & cheap_get_xenos_near just looper over the entire human/xeno lists for that zlevel